subsys/bluetooth/audio/cap_initiator.c

```diff
@@ -1475,6 +1475,8 @@ int bt_cap_initiator_unicast_audio_update(const struct bt_cap_unicast_audio_upda
 	bt_cap_common_set_subproc(BT_CAP_COMMON_SUBPROC_TYPE_META_UPDATE);
 
 	proc_param = get_next_proc_param(active_proc);
+	__ASSERT(proc_param != NULL, "proc is not done, but could not get next proc_param");
```

Comment:
Does this indicate some stack-internal bug, or can this happen if the app passes something invalid to this API? Just trying to understand if `CHECKIF()` would be more appropriate than an assert.

Reply:
Do we have guidelines on that?

Reply:
But it will be checked afterward in the `bt_bap_stream_release` function. Isn't that enough?

Reply:
> Does this indicate some stack-internal bug,

It does indicate a stack-internal bug. The `valid_unicast_audio_update_param` check above prevent `proc_param` from ever being NULL.

Reply:
True. As mentioned in previous comments, at line 1478, the `proc_param` was checked in `valid_unicast_audio_update_param` to prevent it from being null. Similarly, at line 1995, it was checked in `valid_unicast_audio_stop_param`.

Reply:
Hmm, @jhedberg maybe you know, but actually IIRC coverity (at least in the past) has not run with asserts enabled. In which case, adding an `__assert` won't make coverity happy here either.

Reply:
In lines 929, 1132, 1231, and 1313 (of `cap_initiator.c`), it is handled in the same way (using `__ASSERT` after `get_next_proc_param`). So, if this approach is not acceptable, then all should be changed.

Reply:
> In lines 929, 1132, 1231, and 1313 (of `cap_initiator.c`), it is handled in the same way (by `__ASSERT` after `get_next_proc_param`). So, if this approach is not acceptable, then all should be changed.

That's a good point. Let's assume coverity will accept it :)